### ISSUE TITLE
⚠️ Math_Min() and Math_Max() are doing the opposite of what you would expect

### ISSUE THREAD

**Issue by Jay2k1** [bcserv/smlib#70]
In smlib, `Math_Min()` returns the biggest of two given numbers and `Math_Max()` returns the smallest of two given numbers:

https://github.com/bcserv/smlib/blob/aad2c8e963dbd7790096efd7920bd9f3cf76082d/scripting/include/smlib/math.inc#L53-L85

In doing so, they are doing the exact opposite of what anyone who has ever used max and min functions in other languages will expect them to do.

In other programming/scripting languages, the max function returns the maximum value (i.e. the biggest) of a given set and min returns the minimum (the smallest) number of a given set.
I bet that everyone who scripts some SourcePawn and uses the functions provided here will be **very** confused about the results, then start debugging their code and at some point read the source code of the functions - and then be even more confused about why they are "inverted" and pull their hair out.

* [PHP: max — Find highest value](https://www.php.net/manual/en/function.max.php)
* [JS: The Math.max() function returns the largest of the zero or more numbers given as input parameters](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/max)
* [C++: fmax: Returns the larger of two floating point arguments](https://en.cppreference.com/w/c/numeric/math/fmax)
* [Python: max - Return the largest item in an iterable or the largest of two or more arguments.](https://docs.python.org/3/library/functions.html#max)
* [Go: Max returns the larger of x or y.](https://pkg.go.dev/math#Max)
* [Perl: max - Returns the entry in the list with the highest numerical value.](https://metacpan.org/pod/List::Util#max)

**Comment by peace-maker** [comment]
Yes, it's too late to change that behavior now though. :( We could deprecate it and add some `Math_Min2` stock which is gross.

**Comment by Jay2k1** [comment]
I agree to both of that, although I think deprecating them and finding a new naming scheme is preferable to leaving it as it is.  Either way, maybe this issue will pop up on google for people who are looking for a min/max function in SP/SM and help reduce confusion...
